src/Counters.cpp

```diff
@@ -16,36 +16,40 @@
 // this program; if not, write to the Free Software Foundation, Inc., 51
 // Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
 
-// This file contains tests for statistics handling.
+// A simple array of unsigned integers used for the statistics counters.
 
-#include "../src/counters.hpp"
-#include "../src/stats.hpp"
-#include "framework.hpp"
-#include "util.hpp"
+#include "Counters.hpp"
 
-TEST_SUITE(stats)
+#include "stats.hpp"
 
-TEST(forward_compatibility)
+#include <algorithm>
+
+Counters::Counters() : m_counters(0, STATS_END)
```

Comment:
Sorry I don't understand this. Wouldn't this call create 0 elements with value STATS_END?
Why all this complexity: isn't it a fixed size vector (std::array) of STATS_END elements?

Reply:
Thanks, wrong argument order, fixed.

And a vector is required because future ccache versions might add more stats fields which must not get lost when rewritten by an older version, see the added test case.

Reply:
We should convert all remaining enums to enum class in the future to avoid such problems. That's exactly what it has been invented for.

Reply:
Yes, but exactly *here* the automatic decay into an integer is nice to have because of the forward compatibility. Both with `operator[](int)` and `operator[](StatsCounterEnum)` some public `static_cast`ing would be needed. To hide this the counter and the stats functions should probably be jointed so this can be hidden.